### PR TITLE
Cassandra 16758 4.0.0

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
- CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/366/workflows/bd235fe8-9339-42d0-8bcb-9a53e008b383)
- CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/366/workflows/75406e68-d74c-40f4-b717-85838d62890d) (notice failures are from a different/unrelated test method)

**Review by adelapena — APPROVED** [review]
Looks good to me, +1. Perhaps we should change the name of the ticket to something like `Flaky ClientResourceLimitsTest.testBackPressureWhenEndpointLimitExceeded` since there are other flakies in the class that we are not addressing with the patch (`testQueryExecutionWithThrowOnOverload` and `testQueryUpdatesConcurrentMetricsUpdate`)

File: test/unit/org/apache/cassandra/transport/ClientResourceLimitsTest.java [hunk 1]
```diff
@@ -215,9 +215,9 @@ private void backPressureTest(Runnable limitLifter, Consumer<ClientResourceLimit
             // When the client attempts to execute the second query, the backpressure
             // mechanism should report the client connection is paused
             assertTrue(started.await(5, TimeUnit.SECONDS));
-            spinAssertEquals("Timed out after waiting 5 seconds for paused " +
+            spinAssertEquals("Timed out after waiting 10 seconds for paused " +
                              "connections metric to increment due to backpressure",
-                             before + 1, pausedConnections::getValue, 5, TimeUnit.SECONDS);
+                             true, () -> pausedConnections.getValue() >= before + 1, 10, TimeUnit.SECONDS);
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/transport/ClientResourceLimitsTest.java:220]: I think we can leave the stricter check in place if we just avoid doing driver initialization.

See https://github.com/apache/cassandra/pull/1045/files#diff-c801dfb00bc39ce5bf66a5876779350dba35def185588d13f5e492b6d79212f7R77 and https://github.com/apache/cassandra/pull/1045/files#diff-822707a68576bd683b72fde39e19ab8a78ed5a1816d95f5d8a00b3f46ac78d03R419

**Comment by bereng** [comment]
Closed in favor of Caleb's proposal.

File: test/unit/org/apache/cassandra/transport/ClientResourceLimitsTest.java (hunk 1 above)
Comment (bereng) [reply test/unit/org/apache/cassandra/transport/ClientResourceLimitsTest.java:220]: +1

**Pull request closed without merging**